src/components/slider/slider.js

```diff
@@ -0,0 +1,210 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import eventedState from '../../globals/js/mixins/evented-state';
+
+class Slider extends mixin(createComponent, initComponentBySearch, eventedState) {
+  /**
+   * Slider.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as an slider.
+   */
+  constructor(element, options) {
+    super(element, options);
+
+    this.sliderActive = false;
+    this.dragging = false;
+
+    this.track = this.element.querySelector(this.options.selectorTrack);
+    this.filledTrack = this.element.querySelector(this.options.selectorFilledTrack);
+    this.thumb = this.element.querySelector(this.options.selectorThumb);
+    this.input = this.element.querySelector(this.options.selectorInput);
+
+    if (this.element.dataset.sliderInputBox) {
+      this.boundInput = this.element.ownerDocument.querySelector(this.element.dataset.sliderInputBox);
+      this._updateInput();
+      this.boundInput.addEventListener('change', (evt) => { this._handleTextInput(evt); });
+    }
+
+    this._updatePosition();
+
+    this.thumb.addEventListener('mousedown', (evt) => { this._handleMouseDown(evt); });
+    this.element.ownerDocument.addEventListener('mouseup', (evt) => { this._handleMouseUp(evt); });
+    this.element.ownerDocument.addEventListener('mousemove', (evt) => { this._handleMouseMove(evt); });
+    this.thumb.addEventListener('keydown', (evt) => { this._handleKeyDown(evt); });
+    this.track.addEventListener('click', (evt) => { this._handleClick(evt); });
+  }
+
+  _handleMouseDown() {
+    this.sliderActive = true;
+  }
+
+  _handleMouseMove(evt) {
+    if (this.sliderActive === true) {
+      this._updatePosition(evt);
+    }
+  }
+
+  _handleMouseUp() {
+    this.sliderActive = false;
+  }
+
+  _handleKeyDown(evt) {
+    this._updatePosition(evt);
+  }
+
+  _handleClick(evt) {
+    this._updatePosition(evt);
+  }
+
+  _handleTextInput(evt) {
+    this.setValue(evt.target.value);
```

Comment:
I think you can get away with removing these `handle` methods and using their function content inside each event listener:

```js
this.thumb.addEventListener('mousedown', () => {
  // this._handleMouseDown(evt);
  this.sliderActive = true;
});
this.element.ownerDocument.addEventListener('mouseup', () => {
  // this._handleMouseUp(evt);
  this.sliderActive = false;
});
this.element.ownerDocument.addEventListener('mousemove', (evt) => {
  // this._handleMouseMove(evt);
  if (this.sliderActive === true) {
    this._updatePosition(evt);
  }
});
this.thumb.addEventListener('keydown', (evt) => {
  // this._handleKeyDown(evt);
  this._updatePosition(evt);
});
this.track.addEventListener('click', (evt) => {
  // this._handleClick(evt);
  this._updatePosition(evt);
});
```